numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -38,6 +38,26 @@
 #include "npy_cblas.h"
 #include "npy_buffer.h"
 
+
+/*
+ * Define a stack allocated dummy array with only the minimum information set:
+ *   1. The descr, the main field interesting here.
+ *   2. The flags, which are needed for alignment;.
+ *   3. The type is set to NULL and the base is the original array, if this
+ *      is used within a subarray getitem to create a new view, the base
+ *      must be walked until the type is not NULL.
+ *
+ * The following should create errors in debug mode (if deallocated
+ * incorrectly), since base would be incorrectly decref'd as well.
+ */
+#define MAKE_DUMMY_STACK_ARRAY(arr, dummy_array)                    \
+        PyArrayObject_fields __dummy_array_fields;                  \
+        __dummy_array_fields.flags = PyArray_FLAGS(arr);            \
```

Comment:
Any reason for preserving the flags?

Reply:
copyswapn needed the "aligned" one IIRC, but it is not needed in GETITEM/SETITEM I think. So I could set it only in the one place.

Reply:
@seberg did you wan to remove the flags from here and move to copyswapn?

Reply:
Hmm, can do, not that it matters much. The main thing that would matter here is if we decide to leave setitem/getitem the way they are since they hold the GIL for sure.